src/vcSettingsUI.cpp

```diff
@@ -762,80 +803,28 @@ void vcSettingsUI_VisualizationSettings(vcState *pProgramState, vcVisualizationS
-        "settingsVisClassBuilding",
-        "settingsVisClassLowPoint",
-        "settingsVisClassKeyPoint",
-        "settingsVisClassWater",
-        "settingsVisClassRail",
-        "settingsVisClassRoadSurface",
-        "settingsVisClassReserved",
-        "settingsVisClassWireGuard",
-        "settingsVisClassWireConductor",
-        "settingsVisClassTransmissionTower",
-        "settingsVisClassWireStructureConnector",
-        "settingsVisClassBridgeDeck",
-        "settingsVisClassHighNoise"
-      };
-
-      if (ImGui::Button(vcString::Get("settingsVisClassShowAll")))
-      {
-        for (size_t i = 0; i < udLengthOf(pVisualizationSettings->customClassificationToggles); i++)
-          pVisualizationSettings->customClassificationToggles[i] = true;
-      }
-
       ImGui::SameLine();
-      if (ImGui::Button(vcString::Get("settingsVisClassDisableAll")))
-      {
-        for (size_t i = 0; i < udLengthOf(pVisualizationSettings->customClassificationToggles); i++)
-          pVisualizationSettings->customClassificationToggles[i] = false;
-      }
+      if (ImGui::Button(udTempStr("%s##RestoreClassificationColors", vcString::Get("settingsRestoreDefaults"))))
+        memcpy(pProgramState->settings.visualization.customClassificationColors, GeoverseClassificationColours, sizeof(pProgramState->settings.visualization.customClassificationColors));
 
-      for (size_t i = 0; i < udLengthOf(s_customClassifications); i++)
-      {
-        ImGui::PushID((int)i);
-        ImGui::Checkbox("", &pVisualizationSettings->customClassificationToggles[i]);
-        ImGui::PopID();
-        ImGui::SameLine();
-        vcIGSW_ColorPickerU32(vcString::Get(s_customClassifications[i]), &pVisualizationSettings->customClassificationColors[i], ImGuiColorEditFlags_NoAlpha);
-      }
+      for (uint8_t i = 0; i < 19; ++i)
+        vcIGSW_ColorPickerU32(vcSettingsUI_GetClassificationName(pProgramState, i), &pProgramState->settings.visualization.customClassificationColors[i], ImGuiColorEditFlags_NoAlpha);
 
       if (ImGui::TreeNode(vcString::Get("settingsVisClassReservedColours")))
       {
-        for (int i = 19; i < 64; ++i)
-        {
-          ImGui::PushID(i);
-          ImGui::Checkbox("", &pVisualizationSettings->customClassificationToggles[i]);
-          ImGui::PopID();
-          ImGui::SameLine();
-          vcIGSW_ColorPickerU32(udTempStr("%d. %s", i, vcString::Get("settingsVisClassReservedLabels")), &pVisualizationSettings->customClassificationColors[i], ImGuiColorEditFlags_NoAlpha);
-        }
+        for (uint8_t i = 19; i < 64; ++i)
+          vcIGSW_ColorPickerU32(vcSettingsUI_GetClassificationName(pProgramState, i), &pProgramState->settings.visualization.customClassificationColors[i], ImGuiColorEditFlags_NoAlpha);
         ImGui::TreePop();
       }
 
       if (ImGui::TreeNode(vcString::Get("settingsVisClassUserDefinable")))
       {
-        for (int i = 64; i <= 255; ++i)
+        for (int xi = 64; xi <= 255; ++xi)
```

Comment:
It doesn't look like `xi` is used anywhere. Should be able to use `for (uint8_t i =...`

Reply:
Can't be, it will overflow.